src/setup.cpp

```diff
@@ -0,0 +1,42 @@
+#include "setup.h"
+#include "shaders/shader_manager.h"
+#include "window.h"
+#include "aldente.h"
+#include "keyboard.h"
+
+void Setup::setup_shaders()
+{
+	// Load shaders via a shader manager.
+	ShaderManager::create_shader_program("basic");
+	ShaderManager::create_shader_program("skybox");
+	ShaderManager::create_shader_program("shadow");
+	ShaderManager::create_shader_program("debug_shadow");
+	ShaderManager::set_default("basic");
+}
+
+void Setup::setup_callbacks()
+{
+	GLFWwindow *window = Aldente::aldente->get_window();
+	glfwSetErrorCallback(Window::error_callback);
```

Comment:
This is kinda gross... Can't we have each class be responsible for setting its own callback?